src/hexkit/providers/akafka/provider/eventsub.py

```diff
@@ -645,21 +664,26 @@ def __init__(
         self._consumer = consumer
         self._publisher = dlq_publisher
         self._dlq_topic = dlq_topic
-        self._retry_topic = retry_topic
+
+        # In KafkaEventSubscriber, we get the service name from the config. However,
+        # the service name in effect here probably differs -- e.g., a DLQ-specific
+        # service might be running the KafkaDLQSubscriber. So we extract the service
+        # name from the DLQ topic name instead.
+        service_name = service_name_from_dlq_topic(dlq_topic)
+        self._retry_topic = service_name + "-retry"
         self._process_dlq_event = process_dlq_event
 
     async def _publish_to_retry(self, *, event: ExtractedEventInfo) -> None:
         """Publish the event to the retry topic."""
         correlation_id = event.headers["correlation_id"]
-        original_topic = event.headers[ORIGINAL_TOPIC_FIELD]
 
         async with set_correlation_id(correlation_id):
             await self._publisher.publish(
                 payload=event.payload,
                 type_=event.type_,
                 key=event.key,
                 topic=self._retry_topic,
-                headers={ORIGINAL_TOPIC_FIELD: original_topic},
+                headers={ORIGINAL_TOPIC_FIELD: self._dlq_topic.rsplit(".")[0]},
```

Comment:
```suggestion
                headers={ORIGINAL_TOPIC_FIELD: self._dlq_topic.rsplit(".", 1)[0]},
```

Reply:
Thanks. There's also the method at the top that had the same mistake. Fixed